src/ngraph/runtime/intelgpu/intelgpu_backend.cpp

```diff
@@ -350,6 +352,34 @@ bool runtime::intelgpu::IntelGPUBackend::compile(shared_ptr<Function> func)
                                      reversed_axes);
             }
         }
+        else if ("Concat" == op->description())
+        {
+            const size_t cldnn_tensor_dims_lim = 4;
+            const size_t ngraph_tensor_dims = get_input_shape(op, 0).size();
+            const shared_ptr<op::Concat> concat_op = static_pointer_cast<op::Concat>(op);
+            const size_t ngraph_concat_axis = concat_op->get_concatenation_axis();
+            vector<string> inputs;
+
+            if (ngraph_tensor_dims > cldnn_tensor_dims_lim)
+            {
+                throw invalid_argument("Unsupported tensor dimensions (" +
+                                       to_string(ngraph_tensor_dims) + ") for \"" +
+                                       op->description() + "\" operation");
+            }
+            cldnn::concatenation::concatenation_axis cldnn_axis =
+                runtime::intelgpu::IntelGPULayout::get_cldnn_axis(
+                    cldnn_tensor_dims_lim - ngraph_tensor_dims + ngraph_concat_axis);
```

Comment:
`get_input_shape(op, 0).size() - ngraph_concat_axis`
and you will not need in the check above in line 363

Reply:
Will not work.
I.e.:
get_input_shape(op, 0).size() = 2
ngraph_concat_axis = 1
In this case cldnn_axis should equal cldnn::concatenation::along_x.
But you will have concatenation::along_f

Reply:
In case of nobody used `IntelGPULayout::get_cldnn_axis` you could change correspondence for cldnn::id and nGraph::is in this function.